src/components/scene/background.js

```diff
@@ -1,49 +1,229 @@
-      console.warn('Background will not overide predefined environment maps');
+    if (scene.environment &&
+      (scene.environment !== this.cubeRenderTarget.texture || scene.environment !== this.lightProbeTarget.texture)
+    ) {
+      console.warn('Background will not override user defined environment maps');
       return;
     }
 
     if (data.generateEnvironment) {
       scene.environment = this.cubeRenderTarget.texture;
+      this.needsEnvironmentUpdate = true;
     } else {
       scene.environment = null;
     }
+
+    if (this.updateInterval) {
+      clearInterval(this.updateInterval);
+    }
+    if (this.data.environmentUpdateFrequency > 0) {
+      this.updateInterval = setInterval(function () {
+        this.needsEnvironmentUpdate = true;
+      }.bind(this), this.data.environmentUpdateFrequency * 1000); // convert seconds to ms
+    }
   },
 
   tick: function () {
-    if (!this.needsEnvironmentUpdate) return;
     var scene = this.el.object3D;
     var renderer = this.el.renderer;
-    var camera = this.el.camera;
+    var xrSession = renderer.xr.getSession();
+    var frame = this.el.sceneEl.frame;
+    var self = this;
+
+    // Update Light Probe
+    // source: view-source:https://storage.googleapis.com/chromium-webxr-test/r886480/proposals/lighting-estimation.html
+    if (frame && this.xrLightProbe) {
+      var estimate = frame.getLightEstimate(this.xrLightProbe);
+
+      if (estimate) {
+        var intensityScalar =
+          Math.max(1.0,
+            Math.max(estimate.primaryLightIntensity.x,
+              Math.max(estimate.primaryLightIntensity.y,
+                estimate.primaryLightIntensity.z)));
+
+        var probeLight = this.probeLight.components.light.light;
+        probeLight.sh.fromArray(estimate.sphericalHarmonicsCoefficients);
+        probeLight.intensity = 1;
+
+        var directionalLight = this.directionalLight.components.light.light;
+        directionalLight.color.setRGB(
+          estimate.primaryLightIntensity.x / intensityScalar,
+          estimate.primaryLightIntensity.y / intensityScalar,
+          estimate.primaryLightIntensity.z / intensityScalar);
+
+        directionalLight.intensity = intensityScalar;
+        directionalLight.position.copy(estimate.primaryLightDirection);
+      } else {
+        console.log('light estimate not yet available');
+      }
+    }
+
+    if (!this.needsEnvironmentUpdate) {
+      return;
+    }
+
+    if (this.needsLightProbeUpdate) {
+      // wait until the XR Session has started before trying to make
+      // the light probe
+      if (!frame) {
+        return;
+      }
+
+      var gl = renderer.getContext();
+
+      this.needsLightProbeUpdate = false;
+      this.glBinding = new XRWebGLBinding(xrSession, gl);
+      gl.getExtension('EXT_sRGB');
+      gl.getExtension('OES_texture_half_float');
+
+      xrSession.requestLightProbe()
+        .then(function (lightProbe) {
```

Comment:
We van maybe factor out some of this code into component methods, the tick function is getting a tad big